components/collector/src/source_collectors/gitlab/base.py

```diff
@@ -114,16 +113,26 @@ def newer(job1: Job, job2: Job) -> Job:
 
     def _include_entity(self, entity: Entity) -> bool:
         """Return whether to count the job."""
-        return not match_string_or_regular_expression(
-            entity["name"],
-            self._parameter("jobs_to_ignore"),
-        ) and not match_string_or_regular_expression(entity["branch"], self._parameter("refs_to_ignore"))
+        lookback_date = self._lookback_date()
+        return (
+            not match_string_or_regular_expression(
+                entity["name"],
+                self._parameter("jobs_to_ignore"),
+            )
+            and not match_string_or_regular_expression(entity["branch"], self._parameter("refs_to_ignore"))
+            and parse_datetime(entity["build_date"]).date() >= lookback_date
+        )
 
     @staticmethod
     def _build_date(job: Job) -> date:
         """Return the build date of the job."""
         return parse_datetime(job.get("finished_at") or job["created_at"]).date()
 
+    def _lookback_date(self) -> date:
+        """Return the lookback cut-off date."""
+        today = now().date()
```

Comment:
I see that `now()` comes from `shared.utils.date_time`, while the `parse_datetime()` comes from `collector_utilities.date_time`. I guess it's fine that we have separate locations for this code to live, but the `collector_utilities.date_time` is actually reimplementing `shared.utils.date_time.now` in `days_ago` and `days_to_go`. Perhaps good to clean this up, so we don't need to do shotgun surgery at some point

Reply:
I will make a change.